src/Orleans.Core/Messaging/Message.cs

```diff
@@ -1383,6 +1405,9 @@ public static object Deserializer(Type expected, IDeserializationContext context
                 if ((headers & Headers.TRANSACTION_INFO) != Headers.NONE)
                     result.TransactionInfo = SerializationManager.DeserializeInner<ITransactionInfo>(context);
 
+                if ((headers & Headers.ACTIVITY_ID) != Headers.NONE)
+                    result.ActivityId = reader.ReadGuid();
```

Comment:
Given that we're going to implement proper distributed tracing support, should we make this some type other than a guid? eg, something which can also hold the various fields needed by the w3c trace context?

Reply:
hmm. Taken how orthogonal EventSource and Activity are, I think we should have a separate thing for w3c trace context, instead of glob them together. The link between the two is quite weak.

Reply:
I generally agree. It's just that:
* Each header added to `Message` has a cost and the next header we add will need to be some kind of *ExtendedHeaders` header or something else which retains the flexibility of adding more headers.
* In our meeting with Liudmila & Noah, we discussed deriving `ActivityId` from the distributed tracing context - so a separate field might not be required.